x-pack/legacy/plugins/infra/server/routes/inventory_metadata/lib/get_aws_metadata.ts

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { RequestHandlerContext } from 'kibana/server';
+import { InventoryAWSAccount } from '../../../../common/http_api/inventory_meta_api';
+import {
+  InfraMetadataAggregationResponse,
+  InfraMetadataAggregationBucket,
+} from '../../../lib/adapters/framework';
+import { InfraSourceConfiguration } from '../../../lib/sources';
+import { KibanaFramework } from '../../../lib/adapters/framework/kibana_framework_adapter';
+
+export interface AWSInventoryMetadata {
+  accounts: InventoryAWSAccount[];
+  projects: string[];
+  regions: string[];
+}
+
+export const getAWSMetadata = async (
+  framework: KibanaFramework,
+  req: RequestHandlerContext,
+  sourceConfiguration: InfraSourceConfiguration
+): Promise<AWSInventoryMetadata> => {
+  const metricQuery = {
+    allowNoIndices: true,
+    ignoreUnavailable: true,
+    index: sourceConfiguration.metricAlias,
+    body: {
+      query: {
+        bool: {
+          must: [{ match: { 'event.dataset': 'aws.ec2' } }],
```

Comment:
I wonder if we should change this to something based on the inventory type (awsEC2, host, pod, container, awsRDS, etc). Since we know the `id` field we could just query the documents where the `id` field exists. We would need to add `type` to the request body. Here is an example of how to find the fields: https://github.com/elastic/kibana/pull/49983/files#diff-30e3350204be7094aaedc373924f630fR31

We could also potentially use `requiredModules` from the inventory models as well. I don't believe we are actually using that field for anything. https://github.com/elastic/kibana/pull/49983/files#diff-224429e1c97605f7fa8f6b26eb9cf981R16